crowbar_framework/app/models/node_object.rb

```diff
@@ -256,6 +256,32 @@ def intended_role=(value)
     @node["crowbar_wall"]["intended_role"] = value
   end
 
+  def disk_roles=(value)
+    @node["crowbar_wall"] ||= {}
+    @node["crowbar_wall"]["disk_roles"] = value
+  end
+
+  def disk_roles
+    disk_roles = crowbar_wall["disk_roles"] || {}
+
+    # remove devices that were claimed after they role was set
+    disk_roles.each do |d, _role|
+      crowbar_wall["disk_roles"].delete d if crowbar_wall[:claimed_disks].key? d
+    end
+
+    # Check for current unclaimed disks, and add unclaimed disk to
+    # disk_roles hash if it is not there yet.
+    # (New disk could have been added to the node)
+    unclaimed_physical_drives.each do |d, _data|
+      unique_name = unique_device_for(d)
+      unless disk_roles[unique_name]
+        crowbar_wall["disk_roles"] ||= {}
+        crowbar_wall["disk_roles"][unique_name] = ""
+      end
+    end
+    crowbar_wall["disk_roles"] || {}
```

Comment:
They could be just 'grouped' together in the code. Or maybe within a separate module (say, DiskHelpers, with these methods 'private'), included into NodeObject. But whatever works :)